vcr/credential/validator.go

```
@@ -110,6 +114,74 @@ func (d defaultCredentialValidator) Validate(credential vc.VerifiableCredential)
 		return failure("'proof' is required for JSON-LD credentials")
 	}
 
+	// CredentialStatus is not specific to the credential type and the syntax (not status) should be checked here.
+	if err := validateCredentialStatus(credential); err != nil {
+		return failure("invalid credentialStatus: %s", err)
+	}
+
+	return nil
+}
+
+// validateCredentialStatus validates the
+func validateCredentialStatus(credential vc.VerifiableCredential) error {
+	// exit if the credential does not contain a status
+	if credential.CredentialStatus == nil {
+		return nil
+	}
+
+	// get base form of all credentialStatus
+	statuses, err := credential.CredentialStatuses()
+	if err != nil {
+		return err
+	}
+
+	// validate per CredentialStatus.Type
+	for _, credentialStatus := range statuses {
+		// base requirements
+		if credentialStatus.ID.String() == "" {
+			return errors.New("credentialStatus.id is required")
+		}
+		if credentialStatus.Type == "" {
+			return errors.New("credentialStatus.type is required")
+		}
+
+		// only accept StatusList2021EntryType for now
+		if credentialStatus.Type != StatusList2021EntryType {
+			continue
```

Maybe return an error if the status type is not supported?